src/Entities/Payment/PaymentRetrieved/PaymentAuthorized.php

```diff
@@ -8,4 +8,36 @@
 
 final class PaymentAuthorized extends _PaymentWithAuthorizationConfig implements PaymentAuthorizedInterface
 {
+    /**
+     * @var \DateTimeInterface
+     */
+    protected \DateTimeInterface $creditableAt;
+
+    /**
+     * @return mixed[]
+     */
+    protected function casts(): array
+    {
+        return \array_merge_recursive(parent::casts(), [
+            'creditable_at' => \DateTimeInterface::class,
+        ]);
+    }
+
+    /**
+     * @return mixed[]
+     */
+    protected function arrayFields(): array
+    {
+        return \array_merge(parent::arrayFields(), [
+            'creditable_at',
+        ]);
+    }
+
+    /**
+     * @return \DateTimeInterface
+     */
+    public function getCreditableAt(): \DateTimeInterface
+    {
+        return $this->creditableAt;
```

Comment:
shouldn't this be nullable? It's nullable on subsequent payment states so I think it should be here as well.

Reply:
Great catch 🤦🏻